### PR TITLE
Feature/login/clk32 887 login view

### PR DESCRIPTION
**Pull request by dongnh99**
## Prerequisite
- [x] Self-review :monocle_face:
- [x] No Crashes ☠️
---

### Root cause
- // Add Login view

### Solution
- // Explain how you fix it

### Impact
- // Login view

### Screenshots (Optional)
- // Capture affected screens
- Light
Simulator Screen Shot - iPhone 12 Pro - 2022-03-05 at 11 23 42
- Dark
Simulator Screen Shot - iPhone 12 Pro - 2022-03-05 at 11 27 30

### CODE REVIEW

**Comment by namnh92** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

- [ ] I understand that "General.Email" is now existed in Localized service

- [ ] I understand that "General.Password" is now existed in Localized service

- [ ] I understand that "Login.AdvancedServerSettings" is now existed in Localized service

- [ ] I understand that "Login.ForgotPassword?" is now existed in Localized service

- [ ] I understand that "Login.SignIn" is now existed in Localized service

- [ ] I understand that "Login.SignUp" is now existed in Localized service

- [ ] I understand that "Login.SignUp.Suggest" is now existed in Localized service

- [ ] I understand that "Login.Version" is now existed in Localized service

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+	var iconFacebook: Image {
+		AppTheme.shared.imageSet.facebookIcon
+	}
+
+	var backgroundColorView: LinearGradient {
+		colorScheme == .light ? backgroundColorGradient : backgroundColorDark
+	}
+
+	var backgroundSignInBt: LinearGradient {
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Do not abbreviate!
```suggestion
	var backgroundSignInButton: LinearGradient {
```

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		colorScheme == .light ? backgroundColorWhite : backgroundColorGradient
+	}
+
+	var foregroundColorView: Color {
+		colorScheme == .light ? foregroundColorPrimary : foregroundColorWhite
+	}
+
+	var foregroundColorViewBt: Color {
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Do not abbreviate!
```suggestion
	var foregroundColorViewButton: Color {
```

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	var foregroundColorWhite: Color {
+		AppTheme.shared.colorSet.offWhite
+	}
+
+	var foregroundColorPrimary: Color {
+		AppTheme.shared.colorSet.primaryDefault
+	}
+
+	var foregroundColorGradient: LinearGradient {
+		LinearGradient(gradient: Gradient(colors: [.white, .white]), startPoint: .leading, endPoint: .trailing)
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Using AppThem.shared.colorSet. Do not hard code color !

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	var backgroundColorWhite: LinearGradient {
+		LinearGradient(gradient: Gradient(colors: [.white, .white]), startPoint: .leading, endPoint: .trailing)
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Using AppThem.shared.colorSet. Do not hard code color !

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	var backgroundColorDark: LinearGradient {
+		LinearGradient(gradient: Gradient(colors: [.black, .black]), startPoint: .leading, endPoint: .trailing)
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Using AppThem.shared.colorSet. Do not hard code color !

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+					.overlay(
+						RoundedRectangle(cornerRadius: Constant.radius)
+							.stroke(foregroundColorViewBt, lineWidth: Constant.lineWidthBorder))
+
+					Spacer(minLength: Constant.spacerBottomView)
+
+					Text("Login.AppVersion:1.1.1".localized)
+						.font(AppTheme.shared.fontSet.font(style: .placeholder3))
+						.foregroundColor(foregroundColorWhite)
+				}
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Need apply dark/light mode to background/ foreground

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+import CommonUI
+
+private enum Constant {
+	static let spacerTopView = 50.0
+	static let spacerBottomView = 20.0
+	static let spacer = 25.0
+	static let spacerBottom = 45.0
+	static let widthLogo = 160.0
+	static let heightLogo = 120.0
+	static let widthSignInBt = UIScreen.main.bounds.width - 30.0
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Do not abbreviate!
Using padding not using screen width

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+struct LoginContentView: View {
+	@State private var email: String = ""
+	@State private var password: String = ""
+	@State private var editingEmail = false
+	@State private var editingPassword = false
+	@State var inputStyle: TextInputStyle  = .normal
+	@Environment(\.colorScheme) var colorScheme
+
+	var body: some View {
+		NavigationView {
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Subviews do not contain NavigationView

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+			ScrollView {
+				backgroundColorView
+				VStack {
+					Spacer(minLength: Constant.spacerTopView)
+
+					// MARK: - Logo
+					imageLogo
+						.resizable()
+						.aspectRatio(contentMode: .fit)
+						.frame(width: Constant.widthLogo, height: Constant.heightLogo)
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 Using CGSize for short code

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+								inputStyle = .normal
+							} else {
+								inputStyle = .highlighted
+							}
+						})
+						SecureTextField(secureText: $password, inputStyle: $inputStyle, inputIcon: iconLock, placeHolder: "General.Password".localized, keyboardType: .default )
+
+						// MARK: - Signin button
+						Button("Login.SignIn".localized) {}
+						.frame(width: Constant.widthSignInBt, height: Constant.heightBt)
```
Comment (namnh92) [inline ClearKeep/Scenes/Login/Views/LoginContentView.swift]: @dongnh99 : Do not fix width of button. Please using padding

**Review by namnh92 — CHANGES_REQUESTED** [review]

File: ClearKeep/Scenes/Login/Views/LoginContentView.swift (hunk 6 above)
Comment (dongnh99) [reply ClearKeep/Scenes/Login/Views/LoginContentView.swift]: this foreground is white in both dark and light mode

**Review by namnh92 — APPROVED** [review]

**Pull request merged**